app/controllers/historical_percent_changes_controller.rb

```diff
@@ -0,0 +1,29 @@
+class HistoricalPercentChangesController < ApplicationController
+  before_action :set_fixed_datum, only: %i[show update destroy]
+
+  def initialize
+   rent_charge_gateway = RentChargeGateway.new
+   fixed_data_gateway = FixedDataGateway.new
+   @view_historical_percent_changes = ViewHistoricalPercentChanges.new(
+    rent_charge_gateway: rent_charge_gateway, fixed_data_gateway: fixed_data_gateway)
+  end
+  
+   # GET /historical-percent-change
+  def index
+    response = @view_historical_percent_changes.execute
+    json_response(response)
+  end
+
+  # GET /historical-percent-change/:year
+  def show
+    response = @view_historical_percent_changes.percent_changes_for_specific_year(@year)
+    return json_response(response) unless response[:error] == :no_record_found
+  end
+
+  private
+
+  def set_fixed_datum
```

Comment:
Should this be `set_year` rather than `set_fixed_datum`, or vice versa? Typically like to define `set_` methods using the same variable names that they define, just to make it as readable as possible.